common/lib/xmodule/xmodule/tabs.py

```diff
@@ -230,6 +236,61 @@ def from_json(tab_dict):
         return tab_type(tab_dict=tab_dict)
 
 
+class TabFragmentViewMixin(object):
+    """
+    A mixin for tabs that render themselves as web fragments.
+    """
+    fragment_view_name = None
+
+    def __init__(self, tab_dict):
+        super(TabFragmentViewMixin, self).__init__(tab_dict)
+        self._fragment_view = None
+
+    @property
+    def link_func(self):
+        """ Returns a function that returns the course tab's URL. """
+
+        # If a view_name is specified, then use the default link function
+        if self.view_name:
+            return super(TabFragmentViewMixin, self).link_func
+
+        # If not, then use the generic course tab URL
+        def link_func(course, reverse_func):
+            """ Returns a function that returns the course tab's URL. """
+            return reverse_func("course_tab_view", args=[course.id.to_deprecated_string(), self.type])
+
+        return link_func
+
+    @property
+    def url_slug(self):
+        """
+        Returns the slug to be included in this tab's URL.
+        """
+        return "tab/" + self.type
+
+    @property
+    def fragment_view(self):
+        """
+        Renders the web fragment for this tab.
```

Comment:
Yep, that was a bad cut-and-paste error. I've changed it to: "Returns the view that will be used to render the fragment."